src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs

```diff
@@ -178,13 +171,41 @@ public ColumnInfo(string input, string output, int maxNumTerms = ValueToKeyMappi
             public readonly bool TextKeyValues;
 
             protected internal string Terms { get; set; }
-        }
 
-        public const string Summary = "Converts input values (words, numbers, etc.) to index in a dictionary.";
-        public const string UserName = "Term Transform";
-        public const string LoaderSignature = "TermTransform";
-
-        public const string FriendlyName = "To Key";
+            /// <summary>
+            /// Describes how the transformer handles one column pair.
+            /// </summary>
+            /// <param name="input">Name of input column.</param>
+            /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>. Null means <paramref name="input"/> is replaced.</param>
+            /// <param name="maxNumTerms">Maximum number of terms to keep per column when auto-training.</param>
+            /// <param name="sort">How items should be ordered when vectorized. By default, they will be in the order encountered.
+            /// If by value items are sorted according to their default comparison, for example, text sorting will be case sensitive (for example, 'A' then 'Z' then 'a').</param>
+            /// <param name="term">List of terms.</param>
+            /// <param name="textKeyValues">Whether key value metadata should be text, regardless of the actual input type.</param>
+            public ColumnInfo(string input, string output = null,
+                int maxNumTerms = ValueToKeyMappingEstimator.Defaults.MaxNumTerms,
+                SortOrder sort = ValueToKeyMappingEstimator.Defaults.Sort,
+                string[] term = null,
+                bool textKeyValues = false
+                )
+            {
+                Contracts.CheckNonWhiteSpace(input, nameof(input));
+                Input = input;
+                Output = output ?? input;
+                Sort = sort;
+                MaxNumTerms = maxNumTerms;
+                Term = term;
+                TextKeyValues = textKeyValues;
+            }
+        }
+        [BestFriend]
+        internal const string Summary = "Converts input values (words, numbers, etc.) to index in a dictionary.";
```

Comment:
If they are already internal what is the advantage of making them BestFriend?

Reply:
Can't build project without best friend attribute.

---
In reply to: [240377196](https://github.com/dotnet/machinelearning/pull/1811#discussion_r240377196) [](ancestors = 240377196)